Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "PropertyFilePathCtrl.h"
+
+AZ_PUSH_DISABLE_WARNING(4244 4251, "-Wunknown-warning-option")
+#include <QDir>
+#include <QFileDialog>
+#include <QHBoxLayout>
+#include <QLineEdit>
+#include <QMessageBox>
+#include <QToolButton>
+AZ_POP_DISABLE_WARNING
+
+#include <AzCore/Utils/Utils.h>
+
+#include <AzQtComponents/Components/Widgets/BrowseEdit.h>
+#include <AzQtComponents/Components/Widgets/FileDialog.h>
+
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/UI/UICore/WidgetHelpers.h>
+
+namespace AzToolsFramework
+{
+    PropertyFilePathCtrl::PropertyFilePathCtrl(QWidget* parent)
+        : QWidget(parent)
+    {
+        QHBoxLayout* layout = new QHBoxLayout(this);
+
+        m_browseEdit = new AzQtComponents::BrowseEdit(this);
+        m_browseEdit->lineEdit()->setFocusPolicy(Qt::StrongFocus);
+        m_browseEdit->setLineEditReadOnly(true);
+        m_browseEdit->setClearButtonEnabled(true);
+        m_browseEdit->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+        QObject::connect(m_browseEdit, &AzQtComponents::BrowseEdit::attachedButtonTriggered, this, &PropertyFilePathCtrl::OnOpenButtonClicked);
+
+        QToolButton* clearButton = AzQtComponents::LineEdit::getClearButton(m_browseEdit->lineEdit());
+        AZ_Assert(clearButton, "Clear button missing from BrowseEdit");
+        QObject::connect(clearButton, &QToolButton::clicked, this, &PropertyFilePathCtrl::OnClearButtonClicked);
+
+        layout->setContentsMargins(0, 0, 0, 0);
+        layout->setSpacing(2);
+        layout->addWidget(m_browseEdit);
+
+        setLayout(layout);
+
+        setFocusProxy(m_browseEdit->lineEdit());
+        setFocusPolicy(m_browseEdit->lineEdit()->focusPolicy());
+    };
+
+    void PropertyFilePathCtrl::SetFilePath(AZ::IO::Path filePath)
```

Comment:
nit: Should `Path` be passed by const ref here? I don't think this is a view-like type afaik (if we want to pass by value then we could `AZStd::move` it into `m_currentFilePath` and then call `.c_str()` on that

Reply:
Accepting it by value and moving it into the `m_currentPath` variable is efficient.